### PR TITLE
LOK-2477 : New syslog config task set and publish to minion.

### PR DESCRIPTION
**Pull request by MusaidAli**
## Description
New syslog config task set and publish to minion.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2477

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Review by fooker — CHANGES_REQUESTED** [review]
See comments in old PR

**Comment by MusaidAli** [comment]
> See comments in old PR

previous pr comments resolved ,renamed variable.

**Review by cgorantla — COMMENTED** [review]
LGTM!   Syslog listener should be configured based on config received from Inventory, created this task for it https://opennms.atlassian.net/browse/LOK-2532

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+message SyslogConfig {
+  bool new_suspect_on_syslog = 1;
+  bool include_raw_message = 2;
+  ListenerConfig listener_config = 3;
+}
+
+message ListenerConfig {
+  string syslog_address = 1;
+  int32  syslog_port = 2;
```
Comment (smunir-onms) [inline shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto:17]: resolved the posted [comment  ](https://github.com/OpenNMS-Cloud/lokahi/pull/1868#discussion_r1606998282)

Comment (fooker) [inline shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto:17]: `listen_address`

File: shared-lib/protobuf/src/main/proto/syslog-config.proto [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+message SyslogConfig {
+  bool new_suspect_on_syslog = 1;
+  bool include_raw_message = 2;
+  ListenerConfig listener_config = 3;
+}
+
+message ListenerConfig {
+  string syslog_address = 1;
+  int32  syslog_port = 2;
```
Comment (fooker) [inline shared-lib/protobuf/src/main/proto/syslog-config.proto:17]: `listen_address`

**Review by fooker — CHANGES_REQUESTED** [review]

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/ConfigUpdateService.java [hunk 3]
```diff
@@ -64,6 +66,11 @@ private void sendConfigUpdatesToMinion(String tenantId, Long locationId) {
         } catch (Exception e) {
             log.error("Exception while sending flows to Minion", e);
         }
+        try {
+            syslogConfigService.sendSyslogConfigToMinion(tenantId, locationId);
+        } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (MusaidAli) [reply inventory/main/src/main/java/org/opennms/horizon/inventory/service/ConfigUpdateService.java:71]: what is suggestion here .

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/syslogconfig/SyslogConfigBean.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    private String listenAddress;
+    private Integer listenPort;
+    private Boolean newSuspectOnSyslog;
+    private Boolean includeRawMessage;
+    private Integer batchSize;
+    private Integer queueSize;
+    private Integer numThreads;
+    private Integer batchIntervalMs;
+
+    public String getListenAddress() {
```
[the remark this answers is not in the document]
Comment (MusaidAli) [reply inventory/main/src/main/java/org/opennms/horizon/inventory/service/syslogconfig/SyslogConfigBean.java]: resolved.

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto (hunk 1 above)
Comment (MusaidAli) [reply shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto]: removed used in protobuf.same file

File: shared-lib/protobuf/src/main/proto/syslog-config.proto (hunk 2 above)
Comment (MusaidAli) [reply shared-lib/protobuf/src/main/proto/syslog-config.proto]: renamed.

**Pull request closed without merging**